contrib/terraform/openstack/modules/compute/main.tf

```diff
@@ -125,15 +129,20 @@ resource "openstack_compute_instance_v2" "k8s_master_no_etcd" {
   }
 
   security_groups = ["${openstack_compute_secgroup_v2.k8s_master.name}",
+    "${openstack_compute_secgroup_v2.bastion.name}",
     "${openstack_compute_secgroup_v2.k8s.name}",
   ]
 
   metadata = {
     ssh_user         = "${var.ssh_user}"
-    kubespray_groups = "kube-master,k8s-cluster,vault"
+    kubespray_groups = "kube-master,${var.supplementary_master_groups},k8s-cluster,vault"
     depends_on       = "${var.network_id}"
   }
 
+  provisioner "local-exec" {
+    command = "sed s/USER/${var.ssh_user}/ contrib/terraform/openstack/ansible_bastion_template.txt | sed s/BASTION_ADDRESS/${var.number_of_bastions > 0 ? var.k8s_master_fips[0] : var.k8s_master_fips[0]}/ > contrib/terraform/group_vars/no-floating.yml"
```

Comment:
It looks like the conditional `${var.number_of_bastions > 0 ? var.k8s_master_fips[0] : var.k8s_master_fips[0]}` always evaluates to `var.k8s_master_fips[0]`